packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -8,69 +9,56 @@ import contractMaps from '@metamask/contract-metadata';
 import {
   ApprovalType,
   ChainId,
-  ERC20,
-  NetworkType,
-  NetworksTicker,
   ORIGIN_METAMASK,
   convertHexToDecimal,
+  InfuraNetworkType,
+  NetworkType,
   toHex,
+  NetworksTicker,
 } from '@metamask/controller-utils';
 import type {
-  BlockTrackerProxy,
-  NetworkController,
+  NetworkClientConfiguration,
+  NetworkClientId,
   ProviderConfig,
-  ProviderProxy,
-} from '@metamask/network-controller';
-import {
-  defaultState as defaultNetworkState,
-  NetworkClientType,
 } from '@metamask/network-controller';
+import { defaultState as defaultNetworkState } from '@metamask/network-controller';
 import type { PreferencesState } from '@metamask/preferences-controller';
 import { getDefaultPreferencesState } from '@metamask/preferences-controller';
 import nock from 'nock';
 import * as sinon from 'sinon';
+import { v1 as uuidV1 } from 'uuid';
 
-import { FakeBlockTracker } from '../../../tests/fake-block-tracker';
 import { FakeProvider } from '../../../tests/fake-provider';
+import type {
+  ExtractAvailableAction,
+  ExtractAvailableEvent,
+} from '../../base-controller/tests/helpers';
+import {
+  buildCustomNetworkClientConfiguration,
+  buildInfuraNetworkClientConfiguration,
+  buildMockGetNetworkClientById,
+} from '../../network-controller/tests/helpers';
 import { ERC20Standard } from './Standards/ERC20Standard';
 import { ERC1155Standard } from './Standards/NftStandards/ERC1155/ERC1155Standard';
 import { TOKEN_END_POINT_API } from './token-service';
-import type { TokenListState } from './TokenListController';
 import type { Token } from './TokenRatesController';
 import { TokensController } from './TokensController';
-import type { AllowedActions, AllowedEvents } from './TokensController';
+import type { TokensControllerMessenger } from './TokensController';
 
-jest.mock('uuid', () => {
-  return {
-    ...jest.requireActual('uuid'),
-    v1: () => '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d',
-  };
-});
+jest.mock('@ethersproject/contracts');
+jest.mock('uuid');
+jest.mock('./Standards/ERC20Standard');
+jest.mock('./Standards/NftStandards/ERC1155/ERC1155Standard');
```

Comment:
Nit: I see that this is mocked here, but are we using this? I see that in most places, we're mocking the prototype for each method we need mocked, and setting up fake return values that way. Which would work just as well even if we don't include this line.

Though mocking the prototype does seem like an anti pattern to me 🤔. Maybe we would be better off using this mock, and returning a fake instance each time that does what we want, similar to how we're mocking ERC20Standard.

Reply:
Good point! Yeah, I wasn't a fan of mocking the prototype. Jest already lets us mock classes quite easily anyway.

Changed in [c753ae1](https://github.com/MetaMask/core/pull/4247/commits/c753ae12e27ed6f15296cbd8b17af9fb24cfb0f8).